athenahealth/lab_results.go

```
@@ -0,0 +1,303 @@
+	LabResultAttachmentTypeJPG  LabResultAttachmentType = "JPG"
+	LabResultAttachmentTypeJPEG LabResultAttachmentType = "JPEG"
+	LabResultAttachmentTypePDF  LabResultAttachmentType = "PDF"
+	LabResultAttachmentTypeTIF  LabResultAttachmentType = "TIF"
+	LabResultAttachmentTypeTIFF LabResultAttachmentType = "TIFF"
+)
+
+type AddLabResultDocumentOptions struct {
+	// AttachmentContents must be Base64 encoded
+	AttachmentContents io.Reader               `json:"attachmentcontents"`
+	AttachmentType     LabResultAttachmentType `json:"attachmenttype"`
+	InternalNote       *string                 `json:"internalnote"`
+	NoteToPatient      *string                 `json:"notetopatient"`
+	ObservationDate    *time.Time              `json:"observationdate"`
+	OriginalFilename   *string                 `json:"originalfilename"`
+	// 1 = high, 2 = normal
+	Priority    *string `json:"priority"`
+	ResultNotes *string `json:"resultnotes"`
+	// Final, Partial, Pending, Preliminary, Corrected, Cancelled
+	ResultStatus *string `json:"resultstatus"`
+	TieToOrderID *int    `json:"tietoorderid"`
+}
+
+type addLabResultDocumentResponse struct {
+	ErrorMessage string `json:"errormessage"`
+	LabResultID  int    `json:"labresultid"`
+	Success      bool   `json:"success"`
+}
+
+// AddLabResultDocument creates a lab result document record of a specific patient
+//
+// POST /v1/{practiceid}/patients/{patientid}/documents/labresult
+//
+// https://docs.athenahealth.com/api/api-ref/document-type-lab-result#Add-lab-result-document-to-patient's-chart
+func (h *HTTPClient) AddLabResultDocumentReader(ctx context.Context, patientID string, departmentID string, opts *AddLabResultDocumentOptions) (int, error) {
+	form := NewFormURLEncoder()
+
+	form.AddString("departmentid", departmentID)
+
+	if opts != nil {
+		if opts.AttachmentContents != nil {
+			form.AddReader("attachmentcontents", newBase64Reader(opts.AttachmentContents))
+		}
+		if len(opts.AttachmentType) > 0 {
```

We may want to return an error if the attachment type is empty. Might be a good idea to capture any/all client-side errors and join. Checkout my comment here https://github.com/eleanorhealth/go-athenahealth/pull/111#discussion_r1293486940

---

Addressed in above comments